### PR TITLE
Fix kubernetes planner to support multi-deployment application

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Enable pipeline sync for multi-deployments Kubernetes application
```

### CODE REVIEW

File: pkg/app/piped/planner/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -186,39 +187,45 @@ func (p *Planner) Plan(ctx context.Context, in planner.Input) (out planner.Outpu
...
+		if len(templateDiffs) > 0 {
+			progressive = true
+
+			if msg, changed := checkImageChange(templateDiffs); changed {
+				desc = msg
+				return
+			}
+
+			desc = fmt.Sprintf("Sync progressively because pod template of workload %s was changed", w.new.Key.Name)
+			return
```
Comment (khanhtc1202) [inline pkg/app/piped/planner/kubernetes/kubernetes.go:227]: nits, move `progressive = true` to L225 before return statement (along with `desc = ...` ) make this more readable 👀

Comment (nakabonne) [reply pkg/app/piped/planner/kubernetes/kubernetes.go:227]: In that case, it doesn't sync progressively when the image was changed, doesn't it?

Comment (nakabonne) [reply pkg/app/piped/planner/kubernetes/kubernetes.go:227]: This effort is for making the description as readable as possible, you know?

**Comment by nakabonne** [comment]
Looks extremely neat. Thank you.
/lgtm

File: pkg/app/piped/planner/kubernetes/kubernetes.go (hunk 1 above)
Comment (khanhtc1202) [reply pkg/app/piped/planner/kubernetes/kubernetes.go:227]: oh, I missed it 👍 But to me maybe `progressive = true` should be added before line L222 too 🤔 But it's not strong opinion 🙆‍♂️

**Comment by khanhtc1202** [comment]
Way to go 🚀 
/approve

**Pull request merged**